src/test/operators/table_scan_test.cpp

```diff
@@ -1072,4 +1114,155 @@ TEST_P(OperatorsTableScanTest, TwoBigScans) {
+
+  ref_table->get_chunk(ChunkID{0})->finalize();
+  ref_table->get_chunk(ChunkID{1})->finalize();
+  ref_table->get_chunk(ChunkID{0})->set_sorted_by(SortColumnDefinition(ColumnID{0}, SortMode::Ascending));
+  ref_table->get_chunk(ChunkID{1})->set_sorted_by(SortColumnDefinition(ColumnID{0}, SortMode::Ascending));
+  auto table_wrapper = std::make_shared<TableWrapper>(std::move(ref_table));
+  table_wrapper->execute();
+
+  const auto scan_sorted = create_table_scan(table_wrapper, ColumnID{0}, PredicateCondition::LessThanEquals, 3);
+  scan_sorted->execute();
+
+  EXPECT_EQ(scan_sorted->get_output()->chunk_count(), 2);
+  for (const auto& chunk_id : {ChunkID{0}, ChunkID{1}}) {
+    const auto& result_chunk_sorted = scan_sorted->get_output()->get_chunk(chunk_id);
+
+    const auto ref_segment =
+        std::dynamic_pointer_cast<const ReferenceSegment>(result_chunk_sorted->get_segment(ColumnID{0}));
+    EXPECT_TRUE(ref_segment->pos_list()->references_single_chunk());
+
+    const auto& chunk_sorted_by = result_chunk_sorted->sorted_by();
+    ASSERT_FALSE(chunk_sorted_by.empty());
+  }
+}
+
+// Scanning two sorted reference segments. Currently, the sorting information is not passed forward when reference
+// segments do not guarantee that they only reference a single physical chunk (see table_scan.cpp).
+TEST_P(OperatorsTableScanTest, SortedFlagSingleChunkNotGuaranteed) {
+  const auto table = get_int_sorted_op()->table;
+
+  std::shared_ptr<Table> ref_table = std::make_shared<Table>(table->column_definitions(), TableType::References);
+
+  auto pos_list_1 = std::make_shared<RowIDPosList>();
+  pos_list_1->emplace_back(RowID{ChunkID{0}, 1});
+  pos_list_1->emplace_back(RowID{ChunkID{0}, 3});
+
+  const auto segment_1 = std::make_shared<ReferenceSegment>(table, ColumnID{0}, pos_list_1);
+  ref_table->append_chunk({segment_1});
+
+  auto pos_list_2 = std::make_shared<RowIDPosList>();
+  pos_list_2->emplace_back(RowID{ChunkID{1}, 0});
+
+  const auto segment_2 = std::make_shared<ReferenceSegment>(table, ColumnID{0}, pos_list_1);
+  ref_table->append_chunk({segment_2});
+
+  ref_table->get_chunk(ChunkID{0})->finalize();
+  ref_table->get_chunk(ChunkID{1})->finalize();
+  ref_table->get_chunk(ChunkID{0})->set_sorted_by(SortColumnDefinition(ColumnID{0}, SortMode::Ascending));
+  ref_table->get_chunk(ChunkID{1})->set_sorted_by(SortColumnDefinition(ColumnID{0}, SortMode::Ascending));
+  auto table_wrapper = std::make_shared<TableWrapper>(std::move(ref_table));
+  table_wrapper->execute();
+
+  const auto scan_sorted = create_table_scan(table_wrapper, ColumnID{0}, PredicateCondition::LessThanEquals, 1234);
+  scan_sorted->execute();
+
+  EXPECT_EQ(scan_sorted->get_output()->chunk_count(), 2);
+  for (const auto& chunk_id : {ChunkID{0}, ChunkID{1}}) {
+    const auto& result_chunk_sorted = scan_sorted->get_output()->get_chunk(chunk_id);
+    const auto& chunk_sorted_by = result_chunk_sorted->sorted_by();
+    ASSERT_TRUE(chunk_sorted_by.empty());
+  }
+}
+
```

Comment:
(two more open items from last review)

Reply:
Where was this resolved?

Reply:
Oh, I thought you just a comment in a random line to tell about the issues.

What bothers you here?

Reply:
Last time I checked, the previous review had open issues about two comments

Reply:
I forgot to mark one as resolved, the other is done with  c3e5a28.